chain/chain/src/store/mod.rs

```diff
@@ -1592,6 +1599,9 @@ impl<'a> ChainStoreUpdate<'a> {
         outcomes: Vec<ExecutionOutcomeWithId>,
         proofs: Vec<MerklePath>,
     ) {
+        if !self.chain_store.save_tx_outcomes {
+            return;
+        }
```

Comment:
This seemed slightly preferable to [this](https://github.com/near/nearcore/pull/13492/files#diff-8c7dc59b02840dc51c30a5d66e959c2eae194c7b53b6387bbcdc50d8a2cb81e5R1941) 

See "Alternative 2" discussed above for the downsides of pushing the `if`to callsites of `save_outcomes_with_proofs`.